repo/wallet/data.go

```diff
@@ -84,25 +85,27 @@ func PinPhoto(reader io.Reader, fname string, nd *core.IpfsNode, apiHost string)
 		return nil, err
 	}
 	r := bytes.NewReader(b)
-	th, _, err := image.Decode(r)
-	if err != nil {
-		return nil, err
-	}
-	th = imaging.Thumbnail(th, 100, 100, imaging.CatmullRom)
-	thb := new(bytes.Buffer)
-	if err = jpeg.Encode(thb, th, nil); err != nil {
-		return nil, err
-	}
 
 	// rewind source reader for add
 	r.Seek(0, 0)
 
 	// top level directory
 	dirb := uio.NewDirectory(nd.DAG)
 
-	// add the images
+	// add the image
 	addFileToDirectory(dirb, r, fname, nd)
-	addFileToDirectory(dirb, bytes.NewBuffer(thb.Bytes()), "thumb.jpg", nd)
+
+
+	t, err := ioutil.ReadAll(thumb)
+	if err != nil {
+		return nil, err
+	}
+
+	tb := bytes.NewReader(t)
+	tb.Seek(0, 0)
+
+	// add the thumbnail
+	addFileToDirectory(dirb, tb, "thumb.jpg", nd)
```

Comment:
hey dude, nice! don't think you need to create a new reader here, just use thumb direct in addFileToDirectory

Reply:
okay, i guess we didn't need the orig reader copy either